parsec/core/trustchain.py

```diff
@@ -0,0 +1,258 @@
+from typing import Tuple, List, Optional
+from pendulum import Pendulum, now as pendulum_now
+
+from parsec.crypto import VerifyKey
+from parsec.api.protocol import UserID
+from parsec.api.data import (
+    DataError,
+    UserCertificateContent,
+    RevokedUserCertificateContent,
+    DeviceCertificateContent,
+)
+
+
+class TrustchainError(Exception):
+    pass
+
+
+class TrustchainContext:
+    def __init__(self, root_verify_key: VerifyKey, cache_validity: int):
+        self.root_verify_key = root_verify_key
+        self.cache_validity = cache_validity
+        self._users_cache = {}
+        self._devices_cache = {}
+        self._revoked_users_cache = {}
```

Comment:
So the `Trustchain` class isn't used here?

Reply:
No, `Trustchain` class is only defined in the backend to return data for the trustchain field in some commands (this field only provide a subset of the entire trustchain that is useful to validate the certificate provided in the command)